src/qpid/broker/amqp/Incoming.cpp

```diff
@@ -108,7 +109,11 @@ namespace {
 }
 
 DecodingIncoming::DecodingIncoming(pn_link_t* link, Broker& broker, Session& parent, const std::string& source, const std::string& target, const std::string& name)
-    : Incoming(link, broker, parent, source, target, name), sessionPtr(parent.shared_from_this()) {}
```

Comment:
Might be nicer to set in initialiser for consistency (could then also make the attribute a const)